lib/std/ascii.zig

```diff
@@ -1,17 +1,41 @@
-// Does NOT look at the locale the way C89's toupper(3), isspace() et cetera does.
-// I could have taken only a u7 to make this clear, but it would be slower
-// It is my opinion that encodings other than UTF-8 should not be supported.
-//
-// (and 128 bytes is not much to pay).
-// Also does not handle Unicode character classes.
-//
-// https://upload.wikimedia.org/wikipedia/commons/thumb/c/cf/USASCII_code_chart.png/1200px-USASCII_code_chart.png
+//! The 7-bit [ASCII](https://en.wikipedia.org/wiki/ASCII) character encoding standard.
+//!
+//! This is not to be confused with the 8-bit [Extended ASCII](https://en.wikipedia.org/wiki/Extended_ASCII).
+//!
+//! Even though this module concerns itself with 7-bit ASCII,
+//! functions use `u8` as the type instead of `u7` for convenience and compatibility.
+//! Characters outside of the 7-bit range are gracefully handled (e.g. by returning `false`).
+//!
+//! See also: https://en.wikipedia.org/wiki/ASCII#Character_set
 
 const std = @import("std");
+const mem = std.mem;
+const testing = std.testing;
+
+// TODO: remove these after the release of 0.10.0
+const isAlNum = @compileError("deprecated: use isAlphanumeric");
+const eqlIgnoreCase = @compileError("deprecated: use eqlInsensitive");
+const isPrint = @compileError("deprecated: use isPrintable");
+const startsWithIgnoreCase = @compileError("deprecated: use startsWithInsensitive");
+const lessThanIgnoreCase = @compileError("deprecated: use lessThanInsensitive");
+const orderIgnoreCase = @compileError("deprecated: use orderInsensitive");
+const indexOfIgnoreCase = @compileError("deprecated: use indexOfInsensitive");
+const indexOfIgnoreCasePos = @compileError("deprecated: use indexOfInsensitivePos");
+const endsWithIgnoreCase = @compileError("deprecated: use endsWithInsensitive");
+const control_code = @compileError("deprecated: use control");
+const isBlank = @compileError("deprecated: use `char == ' ' or char == '\t'`");
+const isXDigit = @compileError("deprecated: use isHexadecimal");
+const isCntrl = @compileError("deprecated: use isControl");
+const isAlpha = @compileError("deprecated: use isAlphabetic");
+const isSpace = @compileError("deprecated: use isWhitespace");
+const spaces = @compileError("deprecated: use whitespace");
+const isPunct = @compileError("deprecated: write your own function suited to your particular case");
+const isGraph = @compileError("deprecated: use `isPrintable(char) and char != ' '`");
```

Comment:
Could we leave these implementations in for a release cycle with a deprecated comment note?  Having both the old and new names available for a release cycle gives projects and their dependencies time to upgrade without breaking.

Reply:
Hmm, well yeah, that's what I think "deprecation" means too: for some time whatever is deprecated will still work as before but you will get a warning of some kind that at some point the thing will get removed and you are asked to use X instead. In Zig however apparently the definition of "deprecation" is different in that you immediately get an error and get told to use something else. Also, in Zig there are no warnings. How would even do this?
Here are a few examples: https://github.com/ziglang/zig/blob/902df103c6151c257c90de9ba5f29f7f4b9dbea2/lib/std/log.zig#L227
https://github.com/ziglang/zig/blob/360ecc1a2f72967f3a3882b3327e130bdc4e18c0/lib/std/math.zig#L177
And I've simply followed suit and did it that way too.

Reply:
It's a relatively little amount of work in the std library and saves a whole bunch of work/headaches for everyone else.  When you make a breaking change to a library, if it's possible to put the library into a state where both the old/new interfaces are supported, then you get the best of both worlds.  For one, you're not breaking anyone's code right away.  For two, people won't be blocked from moving to the new interface.  If you break the old interface, then you can't move to the new interface until all your dependencies move first.  Having a deprecation period where both are supported solves these issues very nicely.